### PR TITLE
A collection of upgrades for 1.7 release

### PR DESCRIPTION
**Pull request by juliemr**
Bundled a couple of small commits together to do a 1.7.0 release.

@hankduan could you take a look?

### CODE REVIEW

File: lib/locators.js [hunk 1]
```diff
@@ -428,4 +428,27 @@ ProtractorBy.prototype.options = function(optionsDescriptor) {
...
+ *       <span id="span2"></span>
+ *     </>
+ *   </>
+ * </div>
+ * @example
+ * spans = element.all(by.deepCss('span'));
+ * expect(spans.count()).toEqual(3);
+ */
+ProtractorBy.prototype.deepCss = function(selector) {
+  // return webdriver.By.css('* >>> ' + selector);
```
Comment (hankduan) [inline lib/locators.js]: remove this?

**Comment by hankduan** [comment]
LGTM

File: lib/locators.js (hunk 1 above)
Comment (juliemr) [inline lib/locators.js]: Made it into a more useful TODO comment.

**Comment by juliemr** [comment]
Thanks for the quick review!

I'll merge this after the holiday, on Tuesday.

**Comment by juliemr** [comment]
Note - this is causing Travis failures because sauce labs has not yet upgraded their version of chromedriver, so the shadow dom tests fail. All other tests are passing. I'll temporarily turn off those tests.

**Pull request merged**